pkg/keystore/memprovider/memprovider.go

```diff
@@ -0,0 +1,27 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package memprovider
+
+import (
+	"github.com/trustbloc/edge-core/pkg/storage"
+	"github.com/trustbloc/edge-core/pkg/storage/memstore"
+)
+
+// Provider represents an in-memory provider for keystore functionality.
+type Provider struct {
+	store storage.Provider
+}
+
+// New returns a new instance of in-memory provider.
+func New() *Provider {
+	return &Provider{store: memstore.NewProvider()}
```

Comment:
why not use edge-core's memstorage?

Reply:
I agree that here we have an unnecessary level of indirection - keystore's memstorage that wraps the one from edge-core doesn't bring anything new (at least currently) so I will get rid of it.